src/entry.c

```diff
@@ -887,6 +887,7 @@ stumpless_set_entry_message_str_w( struct stumpless_entry *entry,
     if( !new_message ){
       return NULL;
     }
+    new_message_size--; // leave off the NULL character
```

Comment:
Yes, I am going through the commits. Could you tell me about this line ?

Reply:
The wide string conversion functions return a size that includes the NULL terminator, but the entry `message_length` member shouldn't include the NULL terminator in it. Stumpless currently assumes that this string is NULL-terminated though, and specifically the `stumpless_get_entry_message` function does when copying out the message to return. If the NULL terminator is left in on this length, the get entry function will try to read an extra byte at the end because it expects a NULL terminator to be there, which is invalid (and causes valgrind errors).

It's a confusing convention, and the next major version (v3.0) will drop the NULL terminators completely for this reason, but for now this line is necessary to keep everything consistent.